rerun_py/rerun_sdk/rerun/datatypes/mat3x3_ext.py

```diff
@@ -1,21 +1,60 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Sequence
 
 import numpy as np
-import numpy.typing as npt
+import pyarrow as pa
+
+from rerun.error_utils import _send_warning
 
 if TYPE_CHECKING:
-    from . import Mat3x3Like
+    from . import Mat3x3ArrayLike, Mat3x3Like
 
 
 class Mat3x3Ext:
+    def __init__(self: Any, rows: Mat3x3Like | None = None, *, columns: Mat3x3Like | None = None) -> None:
+        from . import Mat3x3
+
+        if rows is not None:
+            if columns is not None:
+                _send_warning("Can't specify both columns and rows of matrix.", 1, recording=None)
+
+            if isinstance(rows, Mat3x3):
+                self.flat_columns = rows.flat_columns
+            else:
+                arr = np.array(rows, dtype=np.float32).reshape(3, 3)
+                self.flat_columns = arr.flatten("F")
+        elif columns is not None:
+            arr = np.array(columns, dtype=np.float32).reshape(3, 3)
+            self.flat_columns = arr.flatten()
+        else:
+            _send_warning("Need to specify either columns or columns of matrix.", 1, recording=None)
+            self.flat_columns = np.identity(3, dtype=np.float32).flatten()
+
     @staticmethod
-    def coeffs__field_converter_override(data: Mat3x3Like) -> npt.NDArray[np.float32]:
+    def native_to_pa_array_override(data: Mat3x3ArrayLike, data_type: pa.DataType) -> pa.Array:
         from . import Mat3x3
 
-        if isinstance(data, Mat3x3):
-            return data.coeffs
+        # Normalize into list of Mat3x3
+        if isinstance(data, Sequence):
+            # single matrix made up of flat float array.
+            if isinstance(data[0], float | int):
```

Comment:
Need to check for empty-list before this point or else these `data[0]`s are going to throw.